### PR TITLE
state.gui (#73)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
In callbacks, you'll be able to write:

```
def on_callback(state: State, ...):
    state.gui.download(file)
```
rename GUi.Methods:

- block_ui => hold_actions
- unblock_ui => resume_actions
- show_notification => notify

and keep:

- navigate => navigate
- download => download

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
Everything looks good. I am just thinking about how we can provide autocomplete for the user on typing state.gui

**Comment by FredLL-Avaiga** [comment]
> Everything looks good. I am just thinking about how we can provide autocomplete for the user on typing state.gui

I can't think of a way to do this ...

File: taipy/gui/__init__.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
 value that is managed by the whole Taipy application.
 """
 
-from .data import _DataAccessor
 from .gui import Gui
+from .state import State
 from .icon import Icon
 from .renderers import Html, Markdown
+from .gui_actions import download, hold_actions, navigate, notify, resume_actions
```
Comment (FabienLelaquais) [inline taipy/gui/__init__.py]: I recall we agreed on `hold_control` and `resume_control`...?

File: taipy/gui/gui.py [hunk 2]
```diff
@@ -736,60 +736,31 @@ def load_config(self, app_config: t.Optional[AppConfig] = None) -> None:
...
+        notification_type: str = "I",
         message: str = "",
         browser_notification: t.Optional[bool] = None,
         duration: t.Optional[int] = None,
     ):
-        """Sends a notification to the user interface.
-
-        Args:
-            type (optional(string)): The notification type. This can be one of `"success"`, `"info"`, `"warning"` or `"error"`.
-                To remove the last notification, set this parameter to the empty string.
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py:761]: If this is optional, I guess you should indicate what the default is...

File: taipy/gui/gui.py [hunk 3]
```diff
@@ -801,28 +772,23 @@ def block_ui(
...
-
-        Args:
-            to: page to navigate to. Should be a valid page identifier. If ommitted, navigates to the root page.
-        """
+    def _navigate(self, to: t.Optional[str] = ""):
         to = to or Gui.__root_page_name
         if to not in self._config.routes:
-            warnings.warn(f"cannot navigate to '{to}' which is not a declared route.")
+            warnings.warn(
+                f'cannot navigate to "{to if to != Gui.__root_page_name else "/"}" which is not a declared route.')
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py]: ->
"Cannot navigate to "{to if to != Gui.__root_page_name else "/"}": unknown page."

File: taipy/gui/gui_actions.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    message: str = "",
+    browser_notification: t.Optional[bool] = None,
+    duration: t.Optional[int] = None,
+):
+    """Sends a notification to the user interface.
+
+    Args:
+        state: the current user state as received in any callback.
+        notification_type (optional(string)): The notification type. This can be one of `"success"`, `"info"`, `"warning"` or `"error"`.
+            To remove the last notification, set this parameter to the empty string.
```
Comment (FabienLelaquais) [inline taipy/gui/gui_actions.py:35]: Same remark on default value.

File: taipy/gui/state.py [hunk 5]
```diff
@@ -12,10 +10,12 @@ class State:
     Attributes:
 
         assign (Callable): allows to set a variable inside a lambda function (state.assign("name", value) is equivalent to state.name = value).
+        gui (Gui): easy access to gui methods.
```
Comment (FabienLelaquais) [inline taipy/gui/state.py]: I think we should remove this...

**Review by FabienLelaquais — APPROVED** [review]
Great!
Added minor doc/api change proposals.

File: taipy/gui/gui.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply taipy/gui/gui.py]: 👍

**Pull request merged**